volumina/sliceSelectorHud.py

```diff
@@ -641,58 +605,39 @@ def _get_label_widget(self, backgroundColor, foregroundColor):
-
-                    # Get the whole dataset (debugging purposes)
-                    # whole_set_roi = Roi(start=TinyVector([0, 0, 0, 0]),
-                    # whole_set_slicing = [slice(int(0), int(i)) for i in [2, 1343, 1022, 1]] + [slice(0,1)]
-                    # set_fromRoi = self.layerStack[lbl].datasources[0].dataSlot.get(whole_set_roi).wait()
-                    # set_fromSlice = self.layerStack[lbl].datasources[0].request(whole_set_slicing).wait()
-                    # set_fromVal = self.layerStack[lbl].datasources[0].dataSlot.value
-
-                    # Equivalent calls to opCacheFixer
-                    # val = self.layerStack[lbl].datasources[0].dataSlot.get(roi).wait()
-                    # val = self.layerStack[lbl].datasources[0].dataSlot(roi.start, roi.stop).wait()
-                    val = self.layerStack[lbl].datasources[0].request(slicing).wait()
-
-                    # This does not call cached values, hence no need to hack the cache. But it is way to slow
-                    # val = self.layerStack[lbl].datasources[0].dataSlot.value[int(t),int(y),int(x),int(z)]
-                finally:
-                    if fixed:
-                        for fx, op in zip(fixed, self.layerStack[lbl].datasources[0].dataSlot.getRealOperator().opPredictionPipeline.innerOperators[0].prediction_cache_gui._innerOps):
-                            op._opCacheFixer._fixed = fx
-                if val == 1:  # indicates, that 'sclicing' is labelled with the label 'lbl'
-                    name = self.layerStack[lbl].name
-                    label = name[name.find("(") + 1:name.find(")")]
-                    color = self.layerStack[lbl].tintColor.name()
-                    self.labelWidget.setStyleSheet(f"color: white;"
-                                                   f"background-color: {color};"
-                                                   f"border: none")
-                    self.labelWidget.setText(f"{label}")
-                    break
+            for layer in self.editor.layerStack:
+                if layer.visible:
+                    layer_id = self.editor.imagepumps[imgView].stackedImageSources._layerToIms[layer]
+                    stack_id = self.editor.imageViews[imgView].scene()._tileProvider._current_stack_id
+                    tile_id = self.editor.imageViews[imgView].scene()._tileProvider.tiling.intersected(QRect(QPoint(x,y), QPoint(x,y)))[0]  # There will be just one tile, since we have just a single point
+
+                    with self.editor.imageViews[imgView].scene()._tileProvider._cache:
+                        image = self.editor.imageViews[imgView].scene()._tileProvider._cache.layer(stack_id, layer_id,
+                                                                                                   tile_id)
+                        if image is not None:
+                            colorVal = image.pixelColor(x%blockSize,y%blockSize)
```

Comment:
Hey @m-novikov, this is how I am getting the data right now. Hacking into tileProvider. As we discussed, these are not the raw dataSources...